src/main/java/org/apache/maven/shared/jar/classes/JarClassesAnalysis.java

```diff
@@ -51,8 +57,18 @@
 @Named
 @SuppressWarnings("checkstyle:MagicNumber")
 public class JarClassesAnalysis {
+
     private final Logger logger = LoggerFactory.getLogger(getClass());
 
+    /**
+     * Constant representing the classes in the root of a Multi-Release JAR file.
+     * Meaning outside of any given META-INF/versions/NN/... entry.
+     */
+    private static final Integer ROOT = 0;
+
+    private static final Pattern CLASS_FILTER_MULTI_RELEASE =
+            Pattern.compile("^META-INF/versions/(\\d{1,2})/.*\\.class$");
```

Comment:
Stupid question, what about resources here? Are they overridden/shadowed as well?

Reply:
see: https://github.com/bndtools/bnd/blob/ee0c55e57824954ddbf251eaca6e3a8d748cec25/biz.aQute.bndlib/src/aQuteSbnd/osgi/JPMSModule.java#L134-L141

Reply:
@michael-o see the Java Spec here:
https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/jar/JarFile.html#getJarEntry(java.lang.String)

so the `getJarEntry` does not restrict it to classes only, so resource *should* work the same way!

Reply:
> @michael-o see the Java Spec here: https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/jar/JarFile.html#getJarEntry(java.lang.String)
> 
> so the `getJarEntry` does not restrict it to classes only, so resource _should_ work the same way!

So limiting to classes is not wise...

Reply:
Is anything left here with PR?

Reply:
The changes requested by @michael-o were added in a second commit.